### ISSUE TITLE
Refine the Posts List block

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#32268]
The Posts List block improves the user-friendliness of the Query block in terms of discovery. But beyond that it is problematic because the underlying Query block (and all of its associated complexity) is exposed upon insertion.

1. It is confusing to click one block in the inserter, and see a totally different block(s) inserted on the canvas.
2. ~Features like pagination will not work correctly on a post or page – this is a feature for templates only~
3. Inheriting the query from the URL is useless in this context, and the current UX feels broken

Here's a quick video highlighting these problems:

https://user-images.githubusercontent.com/846565/119796036-ed618b80-bed0-11eb-97b2-ef5588cf49c8.mp4

As you can see, when the Query inherits from the URL, the front/backend does not match. Also, the pagination doesn't appear on the frontend which feels buggy. It's just a confusing experience all-round.

---

It would be interesting to extend the variation so that it retains its "Post List" name post insertion. ~Since pagination will not work in this context there is scope to simplify the block structure by removing the confusing Query Loop, and restricting which blocks can be inserted inside the posts list.~ In a similar vein the "inherit query from URL" option can be hidden in the Inspector. Even the filtering options can be simplified as we wouldn't need to ask the user to specify which post type to display.

Thinking a little further down the road, a dedicated block for displaying `post`s opens the door to:

* More targeted block pattern suggestions that focus on `post`s.
* Other post-type specific variations to improve the UX
* Hiding the Query block in the post editor altogether
* Optimising the Query block for the template editing experience rather than having to entertain a dual purpose

**Comment by jameskoster** [comment]
Here's a first take on this:

https://user-images.githubusercontent.com/846565/119842525-a9857b00-befe-11eb-82c9-d516ddc3bcf9.mp4

* "Posts List" renamed simply to "Posts"
* Query Loop block is hidden in the UI
* Post type selector is removed
* "Offset" and "Include sticky posts" option moved to "Advanced" panel
* Confusing "inherit query from URL" setting is hidden
* Filtering options tidied up
* Layout and ordering options tidied up (icons need some attention) 

Figma prototype [here](https://www.figma.com/proto/i6yFoDYQdSx8G44thFDd9M/Query-block?page-id=160%3A86&node-id=2529%3A2722&viewport=-1292%2C-1185%2C0.24427157640457153&scaling=scale-down).

**Comment by ntsekouras** [comment]
>Also, the pagination doesn't appear on the frontend which feels buggy. It's just a confusing experience all-round.

This can also be the case for Query blocks that don't have enough posts to show. The pagination in the editor is not calculating anything and is there for positioning and styling reasons.

Related to that would be: https://github.com/WordPress/gutenberg/issues/30369 and https://github.com/WordPress/gutenberg/pull/32230.

This is indeed challenging because the `inherit` option set to `true` is sensible as a default for template editing, while set to `false` is sensible default for single post/page editing. There is also technical challenges with[ innerBlocks](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/nested-blocks-inner-blocks/) like:

>Note: A single block can only contain one InnerBlock component.

**Comment by jameskoster** [comment]
> This can also be the case for Query blocks that don't have enough posts to show. The pagination in the editor is not calculating anything and is there for positioning and styling reasons.

Ah yes, I think that's a separate issue. My point here is that the pagination _will never_ display anything on the frontend in this particular scenario. So it doesn't make much sense for that block to even be available for insertion.

**Comment by jasmussen** [comment]
Coming back to this one, as an excercise I used Figma components to consider an updated "Latest Posts" inspector:
<img width="1336" alt="Latest Posts inspector controls" src="https://user-images.githubusercontent.com/1204802/154940221-66e40270-dc1f-41aa-acaf-1d648502d78d.png">

It's definitely challenged with complexity and could use a lot of work. But there might also be reusable ingredients from the above, and in case it inspires anyone, [here's the figma](https://www.figma.com/file/vmHxwa8WhiyiP9poe2om9J/Inspector-Panels?node-id=12573%3A26814).

**Comment by jameskoster** [comment]
Hey @jasmussen, just clarifying that this issue is focussed on the Posts List block – a Query variant. Latest Posts is something different – an evolution of an old widget iirc.

That said, now I'm wondering if they should be one and the same.

**Comment by jasmussen** [comment]
Yep, the idea was that the inspectors could at least be similar if not the same, as there's presumably going to be a great deal of overlap.

**Comment by jameskoster** [comment]
I suppose it will depend on whether we leverage the Query block for Latest Posts. If we do, then a great deal of these options can be removed in favor of the child blocks inside Post Template.

**Comment by jameskoster** [comment]
Depending on how much control over the query we want to offer, we might provide a 'query builder' interface – based on design tools UX patterns – that handles all manner of filters:

<img width="576" alt="Screenshot 2022-05-09 at 16 19 34" src="https://user-images.githubusercontent.com/846565/167442440-9d2bd4e7-e638-4460-a93e-da756349329e.png">

**Comment by carolinan** [comment]
@jameskoster Can you help me by clarifying in what way the query pagination does not work on a post or page?  It is working in my test.

If you are editing a page that you then set as a static home page, it also makes a lot of sense to have a query pagination.

**Comment by jameskoster** [comment]
Looks like it got fixed :)

**Comment by mtias** [comment]
Is there anything that remains actionable here?

**Comment by jameskoster** [comment]
The Posts List block doesn't seem to exist any more so I guess not 😆
